cachito/workers/tasks/general.py

```diff
@@ -169,6 +171,36 @@ def save_bundle_archive_checksum(request_id: int) -> None:
     bundle_dir.bundle_archive_checksum.write_text(checksum, encoding="utf-8")
 
 
+def _check_packages_data_on_api(
+    request_id: int, packages_count: int, dependencies_count: int
+) -> bool:
+    request = get_request(request_id)
+    packages = request.get("packages", [])
+    dependencies = request.get("dependencies", [])
+
+    return len(packages) == packages_count and len(dependencies) == dependencies_count
```

Comment:
Should we also log the counts when there is a mismatch? Imagine if we one day see a case where all the counts are non-zero but they still do not match o_O